### PR TITLE
feat: add avatar gap api

### PR DESCRIPTION
**Pull request by xrkffgg**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/pull/24330#issuecomment-631851588

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |Avatar add new api `gap` to set the unit distance between left and right sides.           |
| 🇨🇳 Chinese | Avatar 新增 `gap` 来设置字符类型距离左右两侧边界单位像素。           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/avatar/demo/dynamic.md](https://github.com/ant-design/ant-design/blob/add-avatar-sizePadding/components/avatar/demo/dynamic.md)
[View rendered components/avatar/index.en-US.md](https://github.com/ant-design/ant-design/blob/add-avatar-sizePadding/components/avatar/index.en-US.md)
[View rendered components/avatar/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/add-avatar-sizePadding/components/avatar/index.zh-CN.md)

### CODE REVIEW

**Comment by yoyo837** [comment]
哥，录gif软件是什么？

**Comment by xrkffgg** [comment]
> 哥，录gif软件是什么？

Mac 我用 Gifox
Win 我用 ScreenToGif

File: components/avatar/__tests__/__snapshots__/demo.test.js.snap [hunk 1]
```diff
@@ -439,6 +439,7 @@ exports[`renders ./components/avatar/demo/dynamic.md correctly 1`] = `
 <div>
   <span
     class="ant-avatar ant-avatar-lg ant-avatar-circle"
+    gap="4"
```
Comment (afc163) [inline components/avatar/__tests__/__snapshots__/demo.test.js.snap:442]: 不应该有这个。

File: components/avatar/index.tsx [hunk 2]
```diff
@@ -72,20 +77,26 @@ export default class Avatar extends React.Component<AvatarProps, AvatarState> {
...
+      nodeWidth !== 0 &&
+      (this.lastChildrenWidth !== childrenWidth || this.lastNodeWidth !== nodeWidth)
     ) {
-      return;
+      this.lastChildrenWidth = childrenWidth;
+      this.lastNodeWidth = nodeWidth;
+    }
+
+    if (this.lastGap !== gap) {
+      this.lastGap = gap;
```
Comment (afc163) [inline components/avatar/index.tsx:92]: this.lastGap 没什么用。

File: components/avatar/index.zh-CN.md [hunk 3]
```diff
@@ -22,3 +22,4 @@ title: Avatar
 | srcSet | 设置图片类头像响应式资源地址 | string | - |  |
 | alt | 图像无法显示时的替代文本 | string | - |  |
 | onError | 图片加载失败的事件，返回 false 会关闭组件默认的 fallback 行为 | () => boolean | - |  |
+| gap | 字符类型距离左右两侧边界间隙 | number | 4 | 4.3.0 |
```
Comment (afc163) [inline components/avatar/index.zh-CN.md:25]: 单位像素

File: components/avatar/index.tsx [hunk 4]
```diff
@@ -72,20 +77,26 @@ export default class Avatar extends React.Component<AvatarProps, AvatarState> {
...
     // denominator is 0 is no meaning
     if (
-      childrenWidth === 0 ||
-      nodeWidth === 0 ||
-      (this.lastChildrenWidth === childrenWidth && this.lastNodeWidth === nodeWidth)
+      childrenWidth !== 0 &&
+      nodeWidth !== 0 &&
+      (this.lastChildrenWidth !== childrenWidth || this.lastNodeWidth !== nodeWidth)
     ) {
-      return;
```
Comment (xrkffgg) [inline components/avatar/index.tsx:81]: 当 childrenWidth 和 nodeWidth 都不变，只修改 gap 时候，不加这个 lastGap ，就直接 return 了
@afc163

Comment (afc163) [reply components/avatar/index.tsx:82]: lastGap 没有用到吧，我就看到下面有个赋值。

**Review by afc163 — APPROVED** [review]

**Pull request merged**